Microsoft.Research/Contracts/MsCorlib/System.IO.Directory.cs

```diff
@@ -91,7 +91,21 @@ public static String[] GetLogicalDrives()
     }
 #endif
 
-#if !SILVERLIGHT_4_0 && !SILVERLIGHT_5_0
+#if !SILVERLIGHT
+    public static String[] GetFileSystemEntries(string path, string searchPattern, SearchOption searchOption)
+    {
+      Contract.Requires(!String.IsNullOrEmpty(path));
+      Contract.Requires(searchPattern != null);
+      Contract.Requires(searchOption == SearchOption.AllDirectories || searchOption == SearchOption.TopDirectoryOnly);
+      Contract.Ensures(Contract.Result<string[]>() != null);
+      Contract.Ensures(Contract.ForAll(Contract.Result<string[]>(), file => file != null));
+      Contract.EnsuresOnThrow<System.IO.IOException>(true, @"path is a file name.");
```

Comment:
Yeah.. This stuff looked weird for me as well, but then I've searched thought this repo and found that this is a canonical way to express possible exceptions.

My point was more about the message that is absolutely unclear, but because the official documentation is saying that, I'm not sure what we can do. I would prefer to have more clear message but have no idea what this message should looks like.